test/spinxsk/spinxsk.c

```diff
@@ -2067,24 +2062,50 @@ InjectFnmpRxPacket(_In_ FNMP_HANDLE Fnmp) {
                     FNMP_NEIGHBOR_IPV6_ADDRESS, &terminator, &remoteIp.Ipv6) == 0);
         }
 
+        UINT32 payloadLength = 0;
+        if (RandUlong() % 2) {
+            //
+            // Add some QUIC flavor to the TCP / UDP payload.
+            //
+            // We want to generate the QUIC header in the TCP/UDP payload buffer
+            // and we use the frame buffer as a random QUIC payload.
+            //
+            payloadLength = payloadBufferSize;
+            UINT16 quicPayloadLength = (UINT16)(RandUlong() % (payloadBufferSize - QUIC_MAX_HEADER_LEN));
+            UINT8 typeSpecificBits = (UINT8)RandUlong();
+            CONST UINT8 version = 1;
+            UINT8 destConnIdLength = (UINT8)(RandUlong() % XDP_QUIC_MAX_CID_LENGTH);
+            UINT8 destConnId[20];
+            destConnId[0] = (UCHAR)RandUlong();
```

Comment:
so we only set the first byte? can we add a FillBufferRandom() that takes an `_Out_ void *buf, _In_ SIZE_T len`?

Reply:
Good catch, I forgot to fix it after working on the win-net-test side.